client/src/stores/historyStore.ts

```diff
@@ -145,14 +145,21 @@ export const useHistoryStore = defineStore(
             return selectHistory(newHistory as HistorySummary);
         }
 
+        function getNextAvailableHistoryId(excludedIds: string[]) {
+            const historyIds = Object.keys(storedHistories.value);
+            const filteredHistoryIds = historyIds.filter((id) => !excludedIds.includes(id));
+            return filteredHistoryIds[0];
+        }
+
         async function deleteHistory(historyId: string, purge: boolean) {
             const deletedHistory = (await deleteHistoryById(historyId, purge)) as HistorySummary;
-            Vue.delete(storedHistories.value, deletedHistory.id);
```

Comment:
@ElectronicBlueberry The problem was that when history was deleted, the `currentHistory` became undefined and an error was raised until it was replaced in the following.